### ISSUE TITLE
[4.0][a11y]Missing ID attribute in the input field on the Smart Search module

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#23861]
### Steps to reproduce the issue
1. Open the test data page with Smart search module
2. Check the source code

### Expected result
The input field should have an ID attribute so that browsers and assistive technologies recognize the label for this field.

### Actual result
There is a "Search" label for the input field in the Smart Search module. The label can only be visible for screen readers. Unfortunately, the attribute `for` the label refers to an ID that does not exist because the `input` field does not have an ID attribute.  
**Note** When option "Search Field Label" is set to "Show", the `label` is visible, but is not associated with the `input` field.
Chrome and Firefox browsers show in the Accessibility tree that the item has an accessible name, NVDA screen reader announces this name, but testing tools (Audits in Chrome, tota11y, ANDI) report no label.

**Comment by Quy** [comment]
Please test PR #23870<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23861">issues.joomla.org/tracker/joomla-cms/23861</a>.</sub>
